src/Debugger/Test/DebugSessionExtensions.cs

```diff
@@ -23,7 +23,7 @@ internal static class DebugSessionExtensions {
             EventHandler<DebugBrowseEventArgs> handler = null;
             handler = (sender, e) => {
                 session.Browse -= handler;
-                browseTcs.SetResult(true);
+                browseTcs.TrySetResult(true);
```

Comment:
Funnily enough, with this change, it will no longer work. The problem here was that `Browse` was actually raised more than once in some cases, second time being while the first handler is still running, and so `SetResult` was blowing up. So far as I can see, `EventTaskSource` doesn't even try to handle this case. I'm not sure if it should? But I could fix it there and then use it here.